x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -176,8 +197,11 @@ func (k Keeper) SettlePendingClaims(ctx sdk.Context) (
 		// 1. The claim does not require a proof.
 		// 2. The claim requires a proof and a valid proof was found.
 
+		claimApplication := claim.SessionHeader.ApplicationAddress
```

Comment:
nit: in the context of settling a particular claim, there is only one "application", no? Is there a need to disambiguate this application from another? Did I miss something?

```suggestion
		app := claim.GetSessionHeader().GetApplicationAddress()
```

Reply:
Wdyt about adding a TODO here about subscribing to app stake events to increment balances inbetween session settlements?

Reply:
Implemented in #895